thread/src/main/kotlin/io/github/paulgriffith/kindling/thread/model/ThreadDump.kt

```diff
@@ -31,18 +32,27 @@ data class ThreadDump internal constructor(
                 require(lines.size > 2) { "Not a fully formed thread dump" }
                 val firstLine = lines.first()
 
+                val deadlocks: MutableList<Int> = mutableListOf()
+                val startingIndex = if (lines[2].contains("Deadlock")) {
+                    deadlocksPattern.findAll(lines[3]).mapTo(deadlocks) { match -> match.value.toInt() }
+                    5
+                } else 2
+
```

Comment:
Style:
I would drop the `if` expression here. It gets 'overloaded' with the side-effect of mutating deadlocks. I would just do a straightforward if statement and initialize `startingIndex` and `deadlocks` in the two branches.

Also, I like `deadlockIds` better as the field/variable name.

Reply:
Yeah I'll change that, just so it's different from the ThreadDump property name. But the thread dump property name has to match the JSON right? Unless we explicitly give it a different name for serialization to match the JSON thread dumps.